### DISCUSSION TITLE
Please adapt to the wagmi@2.0 version.

### DISCUSSION THREAD

**Discussion by kdtran993** [rainbow-me/rainbowkit#1715] [Ideas]
https://wagmi.sh/react/guides/migrate-from-v1-to-v2

**Comment by o-az** [comment]
it is being worked on. You can track it here https://github.com/rainbow-me/rainbowkit/tree/rainbowkit-wagmi-v2

Feel free to leave a comment on my proposal: https://github.com/rainbow-me/rainbowkit/discussions/1711

**Reply by 0xmkh** [reply]
Appreciate for the response. We've already merged that PR to [beta](https://github.com/rainbow-me/rainbowkit/tree/beta).

**Reply by o-az** [reply]
do you have docs url for the beta version?

**Comment by 0xmkh** [comment]
@kdtran993 We're working toward the launch 🙏, but we do have a `beta` version out right now. Feel free to checkout the migration guide here: https://beta.rainbowkit.com/guides/rainbowkit-wagmi-v2

**Reply by kdtran993** [reply]
Thank you so much.

**Reply by AndonMitev** [reply]
Testing it as well 🙏

**Reply by AndonMitev** [reply]
@KosmosKey hey man about this div data-rk 
<img width="697" alt="image" src="https://github.com/rainbow-me/rainbowkit/assets/32274987/f9d2dd99-b59d-4255-9bcd-546afc3d3530">
what is the role of it? Is it something that can be removed?

**Reply by AndonMitev** [reply]
Also i'm experiencing another issue so i'm on a different network and when hit `Switch Network`, Metamask is not poping up:

<img width="1363" alt="image" src="https://github.com/rainbow-me/rainbowkit/assets/32274987/4544d7d3-7806-4141-b06b-8e966932f1da">

Nothing happens when click to switch the network btw i want to metion WagmiProvider is not utilizing `getDefaultConfig` that is coming from `@rainbow-me/rainbowkit` instead is using https://wagmi.sh/react/guides/ssr wagmi package to create config in order to configure ssr option

**Reply by AndonMitev** [reply]
Hello i've prepared reproducible example: https://codesandbox.io/p/devbox/patient-snowflake-66cckf connect with wallet, then from your wallet change the network to Goerli, try to switch back to Sepolia using the application:

Expected result: To pop up user with wallet to switch the networks
Actual result: Nothing happens after click on Sepolia

**Reply by 0xmkh** [reply]
@AndonMitev This will be fixed with the new chain modal v2 we're going to launch here: https://github.com/rainbow-me/rainbowkit/pull/1709

If you switch your chain which is not included in the dapp it will automatically default back to the first chain you provided to the dapp.

**Reply by 0xmkh** [reply]
@antondalgren We will also introduce ssr in the `getDefaultConfig`. You can always use wagmi's `createConfig `if you have other options to pass in.
